cdap-ui/app/cdap/components/Replicator/List/Deployed/index.tsx

```diff
@@ -128,6 +136,41 @@ const DeployedView: React.FC<WithStyles<typeof styles>> = ({ classes }) => {
 
   React.useEffect(fetchList, []);
 
+  function getPipelineConfig(replicationName) {
+    const replicationObj = objectQuery(configMap, replicationName);
+
+    const replicationConfig = {
+      name: replicationObj.name,
+      artifact: replicationObj.artifact,
+      config: replicationObj.config,
+    };
+
+    return replicationConfig;
+  }
+
+  function exportPipeline(replicationName) {
+    const replicationConfig = getPipelineConfig(replicationName);
+    DownloadFile(replicationConfig);
+  }
+
+  function duplicatePipeline(replicationName) {
+    const replicationObj = getPipelineConfig(replicationName);
+    const replicationConfig = {
+      ...replicationObj,
+      name: '',
+    };
+
+    const cloneId = replicationObj.name;
+    window.localStorage.setItem(cloneId, JSON.stringify(replicationConfig));
```

Comment:
Could we just use the replication pipeline name as cloneid that way we can avoid using the localStorage? It then becomes a sharable URL and the create(or studio in pipelines) can get the pipeline details and create the config. 

Could prevent adding/cleaning up localStorage post-clone.

Reply:
The reason i am using this is so that we have a consistent way between duplicate and import (from a JSON file). If we just send pipeline id in the url, then we'll need 3 separate ways: published pipeline, drafts, and imports.

Reply:
I understand both the operations are creating a new replication pipeline, and you can still have a consistent way, except to do this in create view after fetching the replication pipeline details?

Import
- JSON config imported,
- Config is parsed and validated and navigate to replication pipeline
Clone
- if cloneid exists then fetch the replication pipeline details
- Generate the config from details
- Config is parsed and validated and navigate to replication pipeline

My comment is more about avoiding the use of localStorage (and cleaning up) when we can get the config from backend.

Reply:
Talked to @elfenheart. We need to use localStorage to pass on the config from list to the create view. Discussed possible ways to avoid but for now this is fine.